### ISSUE TITLE
Topic threads in gossip

### ISSUE THREAD

**Issue by pjv** [mikedilger/gossip#223]
So I’m participating in this thread you started: note1445y467yjx73gt5upv3y9y99wm8dnl29jp7mk3tzc34xnnpxm27q48qd5h via gossip and I’m having trouble seeing a lot of individual replies and subthreads branching off of it without doing a lot of gymnastics. In order to see things posted by people i don’t follow i find i have to keep going back and forth between different views in gossip, click the link to the parent post of a reply, or click the little triangle and choose “view thread”. Often I find if I click an author name and then choose “view their posts” that will flush out a lot more of the thread contents and i can then go back to the thread and see more of it fleshed out.

I’m wondering if that’s also your experience with gossip, or if i likely have my relays or settings configured in an odd way.

**Comment by mikedilger** [comment]
Try this:  Go to the parent post and hit the triangle on the right.  That should show all descendants.  If the red bars are highlighting a post further down, you are only subscribed to that posts's descendants... you may see other branches but that's only because they were already in memory.

If that doesn't explain it, then we can talk more about what is going on.

**Comment by pjv** [comment]
Ok, so I’m on 1cda929 at the moment and i have this note at the top of my gossip “thread” window:

<img width="885" alt="SCR-20230215-7l0" src="https://user-images.githubusercontent.com/327716/219015241-2c38d945-ec4a-45a3-92d3-2dee01c60a77.png">

i scroll down through what is shown below and there are 18 total descendants shown at the moment. Three of them are my own posts, several of your replies to me and others. But i know from browsing through that thread yesterday that there are many, many more posts than what i am seeing now. I know i probably posted another 3 or 4 items to that thread. I know that there were whole long lines of other branches descended from that top post that i am not seeing at the moment.

i may just not understand the model of what gossip is and isn’t showing me. I just want to rule out my having this mis-configured.

as i scroll down through the notes, most are black background (dark mode) but some are red background which seems to mean something like it being “new” but the ones I'm looking at in red at the moment are definitely not new to me - i for sure read these in gossip yesterday:

<img width="879" alt="SCR-20230215-7tt" src="https://user-images.githubusercontent.com/327716/219017357-37afd84d-ed9f-4281-b549-9d41bca9ad3b.png">

**Comment by mikedilger** [comment]
Yes ok this is a bug.  I suspected it was.

The 'red' coloring was a dirty hack - it does that whenever the client receives an event from a relay, even if it received it previously from a different relay.  I could clean that up so it only goes red when it is actually new.  But even better is to track if you have seen it yet.

**Comment by pjv** [comment]
i like how the background highlight changes from red to black on mouseover in e97f62a. Can’t remember where i remember that idiom from originally, but it feels nostalgic when i see it happening. 

here’s a potential refinement: i just woke up and did a regossip to get up to level with master. When gossip launched i was at the top of the timeline. first thing i did was scroll to the bottom to read through my follows chronologically. As it was scrolling up with the pointer over the middle of the view, it marked all the notes as read along the way though they were not actually read yet because i was scrolling past them to get somewhere else in the timeline. So… is it possible to have the mouseover/read trigger not fire if scrolling is currently active?

**Comment by mikedilger** [comment]
I concur with your concern.  I was thinking you should have to hover for 1/2 a second or so for it to be marked read.  I don't know if I can detect if scrolling is happening that frame, maybe I can.

**Comment by mikedilger** [comment]
I have pushed a few commits that fix the lion's share of the "missing replies" problem.  It's really a relay-insufficiency problem, in that the relays ought to be returning these replies (we are asking for them), but sometimes they don't.  So I set something up to query the local database just like it queries the relays, and that is returning everything that you might have seen before.

I'm seeing much longer threads now, with replies I never saw before.  I've got a lot of reading to do now.
